velox/connectors/hive/iceberg/IcebergDataSink.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/connectors/hive/HiveDataSink.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+struct IcebergSchema {
+  const std::unordered_map<std::string, std::int32_t> columnNameToIdMapping;
+
+  IcebergSchema(
+      const std::unordered_map<std::string, std::int32_t>&
+          _columnNameToIdMapping)
+      : columnNameToIdMapping(_columnNameToIdMapping) {}
+};
+
+enum TransformType { IDENTITY, YEAR, MONTH, DAY, HOUR, BUCKET, TRUNCATE };
+
+struct IcebergPartitionField {
+  // The ID of the source column in the table schema.
+  int32_t sourceId_;
```

Comment:
The sourceId is the fieldId, I'm not sure if the concept is same between presto and apache iceberg

```
 private Builder year(Types.NestedField sourceColumn, String targetName) {
      checkAndAddPartitionName(targetName);
      PartitionField field =
          new PartitionField(sourceColumn.fieldId(), nextFieldId(), targetName, Transforms.year());
```
Looks like you use the id as inputType index.
```
  PartitionField(int sourceId, int fieldId, String name, Transform<?, ?> transform) {
    this.sourceId = sourceId;
    this.fieldId = fieldId;
    this.name = name;
    this.transform = transform;
  }
```